### PR TITLE
Allow unmarshaling YAML into a Version

### PR DESCRIPTION
**Pull request by dghubble**
* Fixes https://github.com/coreos/bugs/issues/1259

### CODE REVIEW

**Comment by dghubble** [comment]
@crawford

File: semver/semver_test.go [hunk 1]
```diff
@@ -281,6 +283,37 @@ func TestJSON(t *testing.T) {
 	}
 }
 
+func TestYAML(t *testing.T) {
+	document, err := yaml.Marshal(fixtures)
```
Comment (crawford) [inline semver/semver_test.go]: Isn't this just going to marshal into:

```yaml
major: 1
minor: 0
patch: 0
```

Comment (dghubble) [inline semver/semver_test.go]: Default YAML marshalling example:

```
- greaterversion: 1.0.0-alpha.beta
  lesserversion: 1.0.0-alpha.1
- greaterversion: 1.0.0-alpha.1
  lesserversion: 1.0.0-alpha
```

We can implement `MarshalYAML() (interface{}, error)` if we want, but no Ignition tools need marshaling at the moment.

Comment (crawford) [inline semver/semver_test.go]: Huh, I guess that is because we have a `String()`...

**Comment by crawford** [comment]
LGTM

**Pull request merged**